src/main/room/main/G1HRPS/RoomManager.java

```diff
@@ -1,97 +1,305 @@
 package main.G1HRPS;
+private static ArrayList<Room> room_list_;
+/**
+ * A manager class that stores the Room Object as elements in a list. Search methods
+ * parses through the list of rooms, comparing instance variables of each room object with the arguments passed in.
+ * @author LiangHee
+ *
+ */
+public RoomManager() {
 
-import java.util.List;
+}
 
-public class RoomManager implements Supermanager<Room> {
+/**
+ * Takes in a class object and list to add the object into.
+ * @param room This is the room object to be added
+ */
+public static void AddToList(Room room) {
+	room_list_.add(room);
+}
 
-	private List<Room> room_list_;
+/** 
+ * This method takes in input info about room from the menu, creates a new room object, and adds that new object 
+ * to room_list_.
+ * The paramaters are the same as in Room constructor.
+ */
+public static void addNewRoom(RoomType room_type, float price, String room_number, BedSize bedSize, boolean wifiEnabled, boolean withView, boolean Smoking, RoomStatus status){
+	Room room = new Room(room_type, price, room_number, bedSize, wifiEnabled, withView, Smoking, status);
+	AddToList(room);
+}
 
-	public RoomManager() {
-		// TODO - implement RoomManager.RoomManager
-		throw new UnsupportedOperationException();
+/**
+ * Takes in user input from menu and searches for the same string in room number and guestname details of room list.
+ * If same string found, returns object room
+ * @param search_text This is the input that user keys in
+ */
+public Room SearchList(String search_text) {  
```

Comment:
SearchList for all managers, searches the unique identifier (in this case room number) from the list that is managed. i.e. search by room number in list of rooms.

Reply:
In page 1 of the assignment, they wanted room to be able to search room details by both room number/guest, so I added both.